src/db/overrides.rs

```diff
@@ -0,0 +1,126 @@
+use crate::error::Result;
+use postgres::Client;
+use std::time::Duration;
+
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq)]
+pub struct Overrides {
+    pub memory: Option<usize>,
+    pub targets: Option<usize>,
+    pub timeout: Option<Duration>,
+}
+
+impl Overrides {
+    pub fn all(conn: &mut Client) -> Result<Vec<(String, Self)>> {
+        Ok(conn
+            .query("SELECT * FROM sandbox_overrides", &[])?
+            .into_iter()
+            .map(|row| (row.get("crate_name"), Self::from_row(row)))
+            .collect())
+    }
+
+    pub fn for_crate(conn: &mut Client, krate: &str) -> Result<Option<Self>> {
+        Ok(conn
+            .query_opt(
+                "SELECT * FROM sandbox_overrides WHERE crate_name = $1",
+                &[&krate],
+            )?
+            .map(Self::from_row))
+    }
+
+    fn from_row(row: postgres::Row) -> Self {
+        Self {
+            memory: row
+                .get::<_, Option<i64>>("max_memory_bytes")
+                .map(|i| i as usize),
+            targets: row.get::<_, Option<i32>>("max_targets").map(|i| i as usize),
+            timeout: row
+                .get::<_, Option<i32>>("timeout_seconds")
+                .map(|i| Duration::from_secs(i as u64)),
+        }
+    }
+
+    pub fn save(conn: &mut Client, krate: &str, overrides: Self) -> Result<()> {
```

Comment:
can you have this give a warning if we raise the time limit without setting `max_targets` to 1?

Reply:
I added a warning here:

```console
> cargo run -- database limits set serde --timeout 1h
     Running `/home/nemo157/.cargo/shared-target/debug/cratesfyi database limits set serde --timeout 1h`
2023-02-08T16:41:46.520337Z DEBUG docs_rs::db::pool: creating database pool (if this hangs, consider running `docker-compose up -d db s3`)
previous sandbox limit overrides for serde = Some(Overrides { memory: None, targets: None, timeout: Some(3600s) })
2023-02-08T16:41:46.534303Z  WARN docs_rs::db::overrides: setting `Overrides::timeout` implies a default `Overrides::targets = 1`, prefer setting this explicitly
new sandbox limit overrides for serde = Some(Overrides { memory: None, targets: None, timeout: Some(3600s) })
```

I tried also adding a `#[arg(requires = "targets")]` to `timeout` in the CLI interface, but the error that clap gives is not great:

```console
> cargo run -- database limits set serde --timeout 1h
     Running `/home/nemo157/.cargo/shared-target/debug/cratesfyi database limits set serde --timeout 1h`
error: the following required arguments were not provided:
  --targets <TARGETS>

Usage: cratesfyi database limits set --targets <TARGETS> --timeout <TIMEOUT> <CRATE_NAME>

For more information, try '--help'.
```

So I'll leave it with just the runtime warning.